artemis-jms-server/src/main/java/org/apache/activemq/artemis/jms/server/config/impl/ConnectionFactoryConfigurationImpl.java

```diff
@@ -623,6 +636,8 @@ public void decode(final ActiveMQBuffer buffer) {
       deserializationBlackList = BufferHelper.readNullableSimpleStringAsString(buffer);
 
       deserializationWhiteList = BufferHelper.readNullableSimpleStringAsString(buffer);
+
+      enable1xPrefixes = buffer.readBoolean();
```

Comment:
should check buffer has readable bytes before reading, for compatibility if the buffer is written by a previous version

Reply:
@michaelandrepearce  that's correct. I'm surprised the compatibility tests didn't catch this.. probably there's something missing.

Reply:
This is a pretty rare use-case involving JMS connection factories configured and stored on the broker-side.  It's something that Wildfly does for their client/server JNDI implementation.  It's never used by the broker standalone.

Reply:
yes.. but didn't you add a boolean? that would from my quick glance break serialization with older versions.. wouldn't it?